lib/tic_tac_toe.rb

```diff
@@ -7,54 +7,58 @@
 
 class TicTacToe
     attr_accessor :board 
-    attr_reader :player
     def initialize
         @board = Board.new
         @presenter = BoardPresenter.new
+        @players = []
     end
 
-    def player=(input)
-        @player = Player.new(input)
-    end
-
-    def get_player_token
-        token = CommandLineIn.new.get_input.upcase
-        self.player = token
+    def assign_player_token(input)
         input_validator = InputValidator.new
-        if input_validator.valid_token?(player.token)
-            puts "Thanks for selecting #{player.token}. Let's start the game!"
+        if input_validator.valid_token?(input)
+            puts "Thanks for selecting #{input}."
+            player = Player.new(input)
+            @players << player
         else
             puts Prompt::RETRY_MAKE_TOKEN_SELECTION
-            get_player_token
+            selected_token = CommandLineIn.new.get_input.upcase
+            assign_player_token(selected_token)
         end
     end
 
-    def take_turn(board, position)
+    def select_coordinate
+        puts Prompt::MAKE_COORDINATE_SELECTION
+        CommandLineIn.new.get_input.to_i
+    end
+
+    def take_turn(board, position, token)
         input_validator = InputValidator.new
         if input_validator.valid_coordinate?(@board.moves, position)
-            @board.move(player.token, position)
+            @board.move(token, position)
         else
             user_coordinate = select_coordinate
-            take_turn(@board.moves, user_coordinate)
+            take_turn(@board.moves, user_coordinate, token)
```

Comment:
Also, is there a test for this `if` conditional that exists?

Reply:
This is being tested now.